src/server/frontend_wayland/session_lock_v1.cpp

```diff
@@ -227,9 +234,11 @@ void mf::SessionLockV1::destroy()
     if (manager.is_active_lock(this))
     {
         manager.try_relinquish_locking_privilege(this);
+        mir::log_info("SessionLockV1 successfully destroyed itself");
     }
     else
     {
+        mir::log_info("SessionLockV1 attempted to destroy itself when it was not the lock");
```

Comment:
Hm. I wonder if this would be better served by logging *from the wrapper generator* when a protocol error is caught and sent?

Reply:
Eh I kinda like it being optional? Maybe it's not interesting in some contexts?

Reply:
In what contexts do you imagine a protocol error would be uninteresting AND reported by an exception?

It is useful to log a protocol error when it is ignored, but not when it is also reported via an exception.

Reply:
That's fair to me if we think that's useful info. Should this be a separate ticket though to modify the generator to include logging?

Reply:
So basically we don't need the logs here, since we'll get the exception info anyway. Right?